src/h5web/vis-packs/core/matrix/utils.ts

```diff
@@ -1,24 +1,23 @@
 import type { NdArray } from 'ndarray';
 import { hasComplexType, hasNumericType } from '../../../guards';
-import { renderComplex } from '../../../metadata-viewer/utils';
 import type {
   ArrayShape,
   Dataset,
   H5WebComplex,
   Primitive,
 } from '../../../providers/models';
+import { formatMatrixComplex, formatMatrixValue } from '../../../utils';
 import type { PrintableType, ValueFormatter } from '../models';
-import { formatNumber } from '../utils';
 
 export function getFormatter(
   dataset: Dataset<ArrayShape, PrintableType>
 ): ValueFormatter<PrintableType> {
   if (hasComplexType(dataset)) {
-    return (val) => renderComplex(val as H5WebComplex, '.2e');
+    return (val) => formatMatrixComplex(val as H5WebComplex);
```

Comment:
We don't need the recursion behaviour of `renderComplex`, so there's no point in using it here. It means I can use the `formatMatrixComplex` formatter here, and the `formatScalarComplex` formmater in `renderComplex`, which is a lot more explicit than to pass an optional `specifier`.